File: src/main/java/com/pocketempire/config/BuildingConfigLoader.java
package com.pocketempire.config;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.InputStream;
import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class BuildingConfigLoader {
    private static final Map<String, BuildingConfig> configs;

    static {
        configs = loadConfigs();
    }

    private static Map<String, BuildingConfig> loadConfigs() {
        try {
            ObjectMapper mapper = new ObjectMapper();
            InputStream is = BuildingConfigLoader.class.getResourceAsStream("/buildings.json");
            BuildingConfig[] buildings = mapper.readValue(is, BuildingConfig[].class);

            Map<String, BuildingConfig> map = new HashMap<>();
            for (BuildingConfig building : buildings) {
                map.put(building.getName(), building);
            }
            return map;
        } catch (Exception e) {
            e.printStackTrace();
            return Collections.emptyMap();
        }
    }

    public static BuildingConfig getConfig(String name) {
        return configs.get(name);
    }

    public static Collection<BuildingConfig> getAll() {
        return configs.values();
    }
}

File: src/main/kotlin/com/pocketempire/config/BuildingConfigLoader.kt
package com.pocketempire.config

import com.fasterxml.jackson.databind.ObjectMapper

object BuildingConfigLoader {

    private val configs: Map<String, BuildingConfig> = loadConfigs()

    private fun loadConfigs(): Map<String, BuildingConfig> {
        return try {
            val inputStream = requireNotNull(
                BuildingConfigLoader::class.java
                    .getResourceAsStream("/buildings.json")
            ) {
                "Could not find /buildings.json"
            }

            inputStream.use {
                val buildings = ObjectMapper().readValue(
                    it,
                    Array<BuildingConfig>::class.java
                )

                buildings.associateBy { building -> building.name }
            }
        } catch (e: Exception) {
            e.printStackTrace()
            emptyMap()
        }
    }

    @JvmStatic
    fun getConfig(name: String): BuildingConfig? =
        configs[name]

    @JvmStatic
    fun getAll(): Collection<BuildingConfig> =
        configs.values
}